client/my-sites/hosting/staging-site-card/use-staging-sync.ts

```diff
@@ -43,14 +46,21 @@ export const usePushToStagingMutation = (
 export const usePullFromStagingMutation = (
 	productionSiteId: number,
 	stagingSiteId: number,
-	options: UseMutationOptions< PushStagingMutationResponse, PushStagingMutationError >
+	options: UseMutationOptions<
+		PushStagingMutationResponse,
+		PushStagingMutationError,
+		MutationVariables
+	>
 ) => {
 	const mutation = useMutation( {
-		mutationFn: async () =>
-			wp.req.post( {
-				path: `/sites/${ productionSiteId }/staging-site/pull-from-staging/${ stagingSiteId }`,
-				apiNamespace: 'wpcom/v2',
-			} ),
+		mutationFn: async ( options ) =>
+			wp.req.post(
+				{
+					path: `/sites/${ productionSiteId }/staging-site/pull-from-staging/${ stagingSiteId }`,
+					apiNamespace: 'wpcom/v2',
+				},
+				{ sync_options: options }
```

Comment:
I cleaned this name from `sync_options` to just `options` on the REST endpoint side, please adjust it here too.